### PR TITLE
fix(handlers/GUILD_MEMBER_ADD): check if guildMember is undefined

### PR DESCRIPTION
**Pull request by ayntee**

### CODE REVIEW

File: src/handlers/members/GUILD_MEMBER_UPDATE.ts [hunk 1]
```diff
@@ -26,33 +26,34 @@ export async function handleGuildMemberUpdate(data: DiscordGatewayPayload) {
     payload.guild_id,
   );
   await cacheHandlers.set("members", memberStruct.id, memberStruct);
+  if (guildMember) {
+    if (guildMember?.nick !== payload.nick) {
```
Comment (Skillz4Killz) [inline src/handlers/members/GUILD_MEMBER_UPDATE.ts:30]:
```suggestion
    if (guildMember.nick !== payload.nick) {
```

File: src/handlers/members/GUILD_MEMBER_UPDATE.ts [hunk 2]
```diff
@@ -26,33 +26,34 @@ export async function handleGuildMemberUpdate(data: DiscordGatewayPayload) {
     payload.guild_id,
   );
   await cacheHandlers.set("members", memberStruct.id, memberStruct);
+  if (guildMember) {
```
Comment (itohatweb) [inline src/handlers/members/GUILD_MEMBER_UPDATE.ts:29]: just formatting
```suggestion

  if (guildMember) {
```

File: src/handlers/members/GUILD_MEMBER_UPDATE.ts [hunk 3]
```diff
@@ -26,33 +26,34 @@ export async function handleGuildMemberUpdate(data: DiscordGatewayPayload) {
...
+    if (guildMember?.nick !== payload.nick) {
+      eventHandlers.nicknameUpdate?.(
+        guild,
+        memberStruct,
+        payload.nick!,
+        guildMember?.nick,
```
Comment (Skillz4Killz) [inline src/handlers/members/GUILD_MEMBER_UPDATE.ts:36]:
```suggestion
        guildMember.nick,
```

File: src/handlers/members/GUILD_MEMBER_UPDATE.ts [hunk 4]
```diff
@@ -26,33 +26,34 @@ export async function handleGuildMemberUpdate(data: DiscordGatewayPayload) {
...
-      memberStruct,
-      payload.nick!,
-      guildMember?.nick,
-    );
-  }
-
-  if (payload.pending === false && guildMember?.pending === true) {
-    eventHandlers.membershipScreeningPassed?.(guild, memberStruct);
-  }
+    if (payload.pending === false && guildMember?.pending === true) {
```
Comment (itohatweb) [inline src/handlers/members/GUILD_MEMBER_UPDATE.ts:40]:
```suggestion
    if (payload.pending === false && guildMember.pending === true) {
```

File: src/handlers/members/GUILD_MEMBER_UPDATE.ts [hunk 5]
```diff
@@ -26,33 +26,34 @@ export async function handleGuildMemberUpdate(data: DiscordGatewayPayload) {
...
```
Comment (Skillz4Killz) [inline src/handlers/members/GUILD_MEMBER_UPDATE.ts]:
```suggestion
    if (payload.pending === false && guildMember.pending === true) {
```

File: src/handlers/members/GUILD_MEMBER_UPDATE.ts [hunk 6]
```diff
@@ -26,33 +26,34 @@ export async function handleGuildMemberUpdate(data: DiscordGatewayPayload) {
...
+      eventHandlers.membershipScreeningPassed?.(guild, memberStruct);
+    }
 
-  const roleIds = guildMember?.roles || [];
+    const roleIds = guildMember?.roles || [];
```
Comment (Skillz4Killz) [inline src/handlers/members/GUILD_MEMBER_UPDATE.ts:44]:
```suggestion
    const roleIds = guildMember.roles || [];
```

**Review by Skillz4Killz — APPROVED** [review]

**Review by itohatweb — APPROVED** [review]

**Pull request merged**